commit 017e0d502681c84879c743f2fb1baf24231a80c9
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:34 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-headerpaths,pad.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 6fef794b6..50555ed99 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -285,7 +285,7 @@ proc portconfigure::configure_get_cppflags {} {
     if {${compiler.limit_flags}} {
         return {}
     } else {
-        return -I${prefix}/include
+        return -isystem${prefix}/include
     }
 }
 default configure.ldflags       {[portconfigure::configure_get_ldflags]}
@@ -294,7 +294,11 @@ proc portconfigure::configure_get_ldflags {} {
     if {${compiler.limit_flags}} {
         return -Wl,-headerpad_max_install_names
     } else {
-        return "-L${prefix}/lib -Wl,-headerpad_max_install_names"
+        if {[option os.platform] eq "darwin"} {
+            return "-L${prefix}/lib -Wl,-headerpad_max_install_names"
+        } else {
+            return "-L${prefix}/lib -Wl,--enable-new-dtags -Wl,-rpath,${prefix}/lib"
+        }
     }
 }
 default configure.libs          {}
